File: src/backend/CareFirstClinic.API/CareFirstClinic.API/Models/Stock.cs
namespace CareFirstClinic.API.Models
{
    public class Stock
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        public string MedicineName { get; set; } = string.Empty;
        public string? MedicineCode { get; set; }       
        public string? Unit { get; set; }               
        public string? Manufacturer { get; set; }      

        public int Quantity { get; set; }               
        public int MinQuantity { get; set; } = 10;      
        public decimal UnitPrice { get; set; }          

        public bool IsActive { get; set; } = true;

        // Navigation
        public ICollection<PrescriptionDetail> PrescriptionDetails { get; set; }
            = new List<PrescriptionDetail>();
    }
}

File: src/frontend/src/types/stock.ts
export interface Stock {
  id: string;
  medicineName: string;
  medicineCode?: string;
  unit?: string;
  manufacturer?: string;
  quantity: number;
  minQuantity: number;
  unitPrice: number;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateStockDTO {
  medicineName: string;
  medicineCode?: string;
  unit?: string;
  manufacturer?: string;
  quantity: number;
  minQuantity?: number;
  unitPrice: number;
}

export interface UpdateStockDTO {
  medicineName?: string;
  medicineCode?: string;
  unit?: string;
  manufacturer?: string;
  quantity?: number;
  minQuantity?: number;
  unitPrice?: number;
  isActive?: boolean;
}

export interface ImportStockDTO {
  quantity: number;
}